### PR TITLE
Throw more sensical errors in two cases.

### PR DESCRIPTION
**Pull request by zarutian**
Those cases are:
1. when recieving CTP_RETURN for a question not asked at that time or earlier
2. when recieving CTP_RESOLVE for a promise that had not been imported at that time.

Makes debugging issues like #3289 somewhat easier.

### CODE REVIEW

File: packages/captp/lib/captp.js [hunk 1]
```diff
@@ -325,6 +325,9 @@ export function makeCapTP(ourId, rawSend, bootstrapObj = undefined, opts = {}) {
     // Answer to one of our questions.
     async CTP_RETURN(obj) {
       const { result, exception, answerID } = obj;
+      if (!questions.has(answerID)) {
+        throw new Error('Got an answer to a question we have not asked.');
```
Comment (michaelfig) [inline packages/captp/lib/captp.js]: Please include the `answerID` in this error.

File: packages/captp/lib/captp.js [hunk 2]
```diff
@@ -335,6 +338,9 @@ export function makeCapTP(ourId, rawSend, bootstrapObj = undefined, opts = {}) {
     // Resolution to an imported promise
     async CTP_RESOLVE(obj) {
       const { promiseID, res, rej } = obj;
+      if (!imports.has(promiseID)) {
+        throw new Error('Got a resolvement of a promise we have not imported.');
```
Comment (michaelfig) [inline packages/captp/lib/captp.js]: Also, please include promiseID.

**Review by michaelfig — APPROVED** [review]

File: packages/captp/lib/captp.js (hunk 1 above)
Comment (zarutian) [reply packages/captp/lib/captp.js]: Done.

File: packages/captp/lib/captp.js (hunk 2 above)
Comment (zarutian) [reply packages/captp/lib/captp.js]: And done.

**Comment by michaelfig** [comment]
Thanks, @zarutian!

I updated this PR to use template literals, and am merging it now.

**Comment by zarutian** [comment]
Completely forgot that quasiLiterals, now known as template literals, were a thing in javascript now, hence the defensive .concat() use.

Still, havent seen any viable syntax for quasiCodeLiterals yet in js.
(Quite like Scheme quote unquote special forms but more Algol-68 syntax like and for code blocks)

**Pull request merged**